WordPress/src/main/java/org/wordpress/android/ui/accounts/helpers/FetchBlogListWPOrg.java

```diff
@@ -71,154 +78,298 @@ public void fetchBlogList(Callback callback) {
         (new FetchBlogListTask(callback)).executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
     }
 
-    private String getRsdUrl(String baseUrl) throws SSLHandshakeException {
-        String rsdUrl;
-        rsdUrl = ApiHelper.getRSDMetaTagHrefRegEx(baseUrl);
-        if (rsdUrl == null) {
-            rsdUrl = ApiHelper.getRSDMetaTagHref(baseUrl);
-        }
-        return rsdUrl;
-    }
-
     private boolean isHTTPAuthErrorMessage(Exception e) {
-        if (e != null && e.getMessage() != null && e.getMessage().contains("401")) {
-            mHttpAuthRequired = true;
-            return true;
-        }
-        return false;
+        return e != null && e.getMessage() != null && e.getMessage().contains("401");
     }
 
-    private String getXmlrpcByUserEnteredPath(String baseUrl) {
-        String xmlRpcUrl;
-        if (!UrlUtils.isValidUrlAndHostNotNull(baseUrl)) {
-            AppLog.e(T.NUX, "invalid URL: " + baseUrl);
+    private Object doSystemListMethodsXMLRPC(String url) throws XMLRPCException, IOException, XmlPullParserException {
+        if (!UrlUtils.isValidUrlAndHostNotNull(url)) {
+            AppLog.e(T.NUX, "invalid URL: " + url);
             mErrorMsgId = org.wordpress.android.R.string.invalid_site_url_message;
             return null;
         }
-        URI uri = URI.create(baseUrl);
+
+        AppLog.i(T.NUX, "Trying system.listMethods on the following URL: " + url);
+        URI uri = URI.create(url);
         XMLRPCClientInterface client = XMLRPCFactory.instantiate(uri, mHttpUsername, mHttpPassword);
-        try {
-            client.call(Method.LIST_METHODS);
-            xmlRpcUrl = baseUrl;
-            return xmlRpcUrl;
-        } catch (XMLRPCException e) {
-            AppLog.i(T.NUX, "system.listMethods failed on: " + baseUrl);
-            if (isHTTPAuthErrorMessage(e)) {
-                return null;
+        return client.call(Method.LIST_METHODS);
+    }
+
+    private boolean validateListMethodsResponse(Object[] availableMethods) {
+        if (availableMethods == null) {
+            AppLog.e(T.NUX, "The response of system.listMethods was empty!");
+            return false;
+        }
+        // validate xmlrpc methods
+        String[] requiredMethods =  { "wp.getUsersBlogs", "wp.getPage", "wp.getCommentStatusList", "wp.newComment",
+                "wp.editComment", "wp.deleteComment", "wp.getComments",	"wp.getComment",
+                "wp.getOptions", "wp.uploadFile", "wp.newCategory",
+                "wp.getTags", "wp.getCategories", "wp.editPage", "wp.deletePage",
+                "wp.newPage", "wp.getPages" };
+
+        for (String currentRequiredMethod: requiredMethods) {
+            boolean match = false;
+            for (Object currentAvailableMethod: availableMethods) {
+                if ((currentAvailableMethod).equals(currentRequiredMethod)) {
+                    match = true;
+                    continue;
```

Comment:
Getting a compiler warning here: `"continue" is unnecessary as the last statement in a loop`